index_documents.go

```diff
@@ -70,26 +39,30 @@ func (i Index) addDocuments(documentsPtr interface{}, contentType string, primar
 		i.PrimaryKey = primaryKey[0] //nolint:golint,staticcheck
 		endpoint = "/indexes/" + i.UID + "/documents?primaryKey=" + primaryKey[0]
 	}
+
+	function := ""
+	if httpMethod == http.MethodPost {
+		function = "Add"
+	} else if httpMethod == http.MethodPut {
+		function = "Update"
+	}
+
 	req := internalRequest{
 		endpoint:            endpoint,
-		method:              http.MethodPost,
+		method:              httpMethod,
 		contentType:         contentType,
 		withRequest:         documentsPtr,
 		withResponse:        resp,
 		acceptedStatusCodes: []int{http.StatusAccepted},
-		functionName:        "AddDocuments",
+		functionName:        function + "Documents",
 	}
 	if err = i.client.executeRequest(req); err != nil {
 		return nil, err
 	}
 	return resp, nil
 }
```

Comment:
Sorry, I realized that I may not have been clear. I was not talking about refactoring the code of all the functions; for me this method (`saveDocuments`) only makes the understanding of the code more complex
I was talking about lines 367 to 398 (previously 408 to 452) which are always duplicated and could be put in the same function.

Reply:
Check it out if that's what you meant

Reply:
Sorry for the delay.
That's right, except you don't have to rewrite the `addDocuments()` method, we don't need the `saveDocument()`.